### PR TITLE
Fix/mobility path

### PR DESCRIPTION
**Pull request by adev4a**
# PR Details
Investigate and fix usdot-fhwa-stol/carma-platform#885
## Description
This branch extends the functionality of the cpp message to accept any input for MobilityPath messages and parses it into required format for encoding and decoding.

During integration testing for MobilityOperations it was found that the input had to be in a certain format for encoding decoding to work. Here, that requirement was explored and removed for MobilityPath messages, to allow strings and numbers of any length to be given as input.
However for encoding, a certain format is required, and the node parses the input into that format.
A warning is issued, and encoding takes place with default input
## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: carma-messenger-core/cpp_message/src/MobilityPath_Message.cpp [hunk 1]
```diff
@@ -45,7 +45,7 @@ namespace cpp_message
             uint64_t timestamp;
             //get sender id
             size_t str_len=message->value.choice.TestMessage02.header.hostStaticId.size;
-            if(str_len<=Header_constant.STATIC_ID_MAX_LENGTH && str_len!=0)
+            if(str_len<=Header_constant.STATIC_ID_MAX_LENGTH && str_len>=Header_constant.STATIC_ID_MIN_LENGTH)
```
Comment (kjrush) [inline carma-messenger-core/cpp_message/src/MobilityPath_Message.cpp:48]: Is static ID variable in size? Or is it a fixed length in the encoding standard? This is fine if it's variable length, but padding may be required if it's fixed.

File: carma-messenger-core/cpp_message/src/MobilityPath_Message.cpp [hunk 2]
```diff
@@ -177,63 +182,99 @@ namespace cpp_message
...
-        MessageFrame_t* message;
-        message=(MessageFrame_t*)calloc(1,sizeof(MessageFrame_t));
+        std::shared_ptr<MessageFrame_t>message_shared(new MessageFrame_t);
         //if mem allocation fails
-        if(!message)
+        if(!message_shared)
         {
-            ROS_WARN_STREAM("Cannot allocate mem for MobilityPath message encoding");
-            return boost::optional<std::vector<uint8_t>>{};
+            ROS_WARN_STREAM("Cannot allocate mem for MobilityRequest message encoding");
```
Comment (kjrush) [inline carma-messenger-core/cpp_message/src/MobilityPath_Message.cpp:192]: Is this supposed to say MobilityPath?

Comment (adev4a) [reply carma-messenger-core/cpp_message/src/MobilityPath_Message.cpp:192]: Yes, this should be mobilitypath

File: carma-messenger-core/cpp_message/src/MobilityPath_Message.cpp (hunk 1 above)
Comment (adev4a) [reply carma-messenger-core/cpp_message/src/MobilityPath_Message.cpp:48]: Static ID is fixed, but the minimum length here is 2, If anything smaller than that is given as input, it assigns a default value of "[ ]"

**Review by kjrush — APPROVED** [review]

**Pull request merged**